### ISSUE TITLE
Question: Which cards to show when doing the daily review

### ISSUE THREAD

**Issue by gitStanda** [open-spaced-repetition/dart-fsrs#17]
Hi everyone! I'm currently building a card review system and having trouble deciding which cards to show and tracking progress for the day. Here's the problem I'm facing:

## My Goals:
- Show cards when doing a review session, right now I show:
    - All cards that are not in newState where due < Time.now().
    - X random cards from the newState pool (X = dailyNewCardsLimit)
- Show users their progress for the day (e.g., cards studied today / total cards to study today ; for each deck).
## My Concerns:
The current implementation seems problematic because due is accurate to the second. This means that during a review session, it might falsely indicate that all reviews are complete, even though a new card could become due just a few minutes later.

## My Specific Questions:
**1) What cards should I show during the review session?**
My current thinking: Show all cards in _(re)learning_ state, regardless of due, and all _review_ state cards that are due, plus somehow X random _newState_ cards.

**2) How do I calculate the cards that were studied today?**
Idea: `cardsStudiedToday = card.lastReview after today_start AND card.due after today_end`. Is this correct or is there a better way?

**3) How do I calculate the total cards to be studied today?**
Idea: `totalCardsToStudyToday = cardsStudiedToday + dailyNewCardsLimit + dueCards (due < today_end AND state != newState)`. 

**4) How should I handle time zones?**
Should I care about time zones when calculating what’s “due today”? All of this is happening on the client side, so I wonder if there’s something I’m overlooking.

Sorry if these are stupid questions, I've tried looking at the `ts-fsrs-demo` and other repos, but I'm struggling to fully grasp the concepts.  I can't wrap my head around this 😕 Any guidance or resources would be greatly appreciated! It would be awesome if there was a general guide for how to use FSRS.

Thanks so much in advance for any help

**Comment by gitStanda** [comment]
Btw I'm not sure the learning and relearning means it should be learned that day/session, that's why I'm not sure if I should put it inside the review session regardless of due or not.

**Comment by gitStanda** [comment]
Okay, I am now 99% sure that I should show all **(re)learning** cards + **Review** _where due<Time.now()_. This basically solves question 1.

Plus I'll need to show the **newState** cards. From my previous question, I deducted the simplest way to show them would be to count _number of new cards studied today_ and keep showing new cards until _count >= dailyLimit_. 

Sadly, the only way to count the new cards studied today I know is to store ReviewLog and use that to count it.
This will grow my database by quite a lot. 

Another Idea I had is to create table with 2 columns - date & newCardsStudiedCount. 
Basically, after every review I would check if the card's state is newState and if it is, check for date column with Date.now() - or create one - and raise the count +1 there... this would mean smaller database, but I might need review log later anyway & still sounds more complicated than it has to be. I'm open to any other ideas.

I also gave up on the idea of showing daily progress (question 2 + 3), it's not part of MVP, I can just show "cards remaining for right now"
